tools/pipelines/templates/include-test-real-service.yml

```diff
@@ -278,19 +278,24 @@ jobs:
         - task: Bash@3
           displayName: Unpack test files
           inputs:
-            workingDir: ${{ parameters.testWorkspace }}
+            workingDirectory: ${{ parameters.testWorkspace }}/node_modules/${{ parameters.testPackage }}
             targetType: 'inline'
             script: |
-              mkdir ${{ parameters.testWorkspace }}/node_modules/${{ parameters.testPackage }}/dist/test
-              mkdir ${{ parameters.testWorkspace }}/node_modules/${{ parameters.testPackage }}/src/test
-              tar -xvf $(Pipeline.Workspace)/test-files/${{ parameters.testFileTarName }}.test-files.tar -C $(Pipeline.Workspace)/test-files
-              mv $(Pipeline.Workspace)/test-files/dist/test/* ${{ parameters.testWorkspace }}/node_modules/${{ parameters.testPackage }}/dist/test
-              mv $(Pipeline.Workspace)/test-files/src/test/* ${{ parameters.testWorkspace }}/node_modules/${{ parameters.testPackage }}/src/test
+              TAR_PATH=$(Pipeline.Workspace)/test-files/${{ parameters.testFileTarName }}.test-files.tar
+              echo "Unpacking test files for ${{ parameters.testPackage }} from file '$TAR_PATH' in '$(pwd)'"
+              # Note: we could skip the last argument and have it unpack everything at once, but if we later change
+              # the structure/contents of the tests tar file, the extraction could overwrite things we didn't intend to,
+              # so keeping the paths to extract explicit.
+              # Also, extracting is finicky with the exact format of the last argument, it needs to match how the
+              # tarfile was created (e.g. './lib/test' works here but 'lib/test' does not).
+              tar --extract --verbose --file $TAR_PATH ./lib/test
+              tar --extract --verbose --file $TAR_PATH ./dist/test
+              tar --extract --verbose --file $TAR_PATH ./src/test
 
         - task: Bash@3
           displayName: Copy devDependencies
           inputs:
-            workingDir: ${{ parameters.testWorkspace }}
+            workingDirectory: ${{ parameters.testWorkspace }}
```

Comment:
Misc fix, the previous name is incorrect (also fixed in the bash step above). Kudos to the Azure Pipelines VS Code extension for the squigglies.